core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -461,6 +464,12 @@ public void onLeadershipChange(Set<Partition> partitionsBecomeLeader,
         }
     }
 
+    public void stopPartitions(Set<StopPartition> stopPartitions,
+                               BiConsumer<TopicPartition, Throwable> errorHandler) {
+        // null means remoteLogDisablePolicy is not applied
+        stopPartitions(stopPartitions, errorHandler, null);
```

Comment:
Instead of `null`, can we supply the default value (retain)?

Reply:
I was thinking about it and tracing the callers of `stopPartitions`, I found we invoked it in different cases:
1. stopReplica request
2. stray logs handler
3. apply topic image delta 
4. `applyLocalFollowersDelta` when controller shutdown

In different cases, we can't just expect they want to "retain" the remote logs. Like in "topic deletion" case, we expect all remote logs are deleted and all remote log tasks are cancelled (and we already did it). So we can't default them as "retain" or "delete" here, we need a "default" value here, that is not related to any `remoteLogDisablePolicy`. 

Does that make sense?

Reply:
Got it. Thanks for the clarification!